udata/tests/reuse/test_reuse_model.py

```diff
@@ -75,3 +77,9 @@ def test_reuse_metrics(self):
         assert reuse.get_metrics()['issues'] == 1
         assert reuse.get_metrics()['discussions'] == 1
 
+        with assert_emit(Reuse.on_update):
+            reuse.datasets.append(dataset)
+            reuse.save()
+        
+        reuse.count_datasets()
+        assert reuse.get_metrics()['datasets'] == 2
```

Comment:
Also test dataset.count_reuses() via the job when a dataset is deleted from a reuse.